### PR TITLE
New Quirk: Proper

### PR DESCRIPTION
**Pull request by StrangeWeirdKitten**
## About The Pull Request

New Quirk: Proper

Costs 0 Points: Neautral Quirk
You can wear Latex Heels without falling over
Reduces the slowdown speed for Latex Heels to 1 and Dominant Heels to 0

## How This Contributes To The Skyrat Roleplay Experience

image

## Changelog

:cl: Strange Weird Kitten
add: New Quirk: Proper. You can know wear latex heels without falling over + w/ reduced slowdown
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/lewd_shoes.dm [hunk 1]
```diff
@@ -30,8 +30,16 @@
...
-	if(src == affected_mob.shoes)
+	if(HAS_TRAIT(affected_mob, TRAIT_PROPER)) //Janky workaround for slowdown not updating before if(affected_mob.shoes)
+		slowdown = 1
+	else
+		slowdown = 4
+
+	if((src == affected_mob.shoes) && (!HAS_TRAIT(affected_mob, TRAIT_PROPER))) //Does not process if player has TRAIT_PROPER
 		START_PROCESSING(SSobj, src)
+	else if(src == affected_mob.shoes)
+		to_chat(affected_mob, span_notice("Wearing these heels is almost like second nature"))
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/lewd_shoes.dm:41]: this doesn't feel needed, also add a period if this stays

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/lewd_shoes.dm [hunk 2]
```diff
@@ -114,7 +121,14 @@
 	worn_icon_digi = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_shoes_digi.dmi'
 	equip_delay_other = 60
 	strip_delay = 60
-	slowdown = 1
+
+/obj/item/clothing/shoes/dominaheels/equipped(mob/user, slot)
+	. = ..()
+	if(HAS_TRAIT(user, TRAIT_PROPER))
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/lewd_shoes.dm:127]: make this a ternary

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/lewd_shoes.dm [hunk 3]
```diff
@@ -30,8 +30,16 @@
 /obj/item/clothing/shoes/latexheels/equipped(mob/user, slot)
 	. = ..()
 	var/mob/living/carbon/human/affected_mob = user
-	if(src == affected_mob.shoes)
+	if(HAS_TRAIT(affected_mob, TRAIT_PROPER)) //Janky workaround for slowdown not updating before if(affected_mob.shoes)
+		slowdown = 1
```
Comment (Zonespace27) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/lewd_shoes.dm:34]: ditto

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 4]
```diff
@@ -134,3 +134,14 @@
 
 	var/obj/item/organ/internal/tongue/dog/new_tongue = new(get_turf(human_holder))
 	new_tongue.Insert(human_holder)
+
+/datum/quirk/proper
+	name = "Proper"
+	desc = "You can walk in heels without penalty"
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm:140]: add a period to all of the sentences here

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
I don't quite see the reason for having this at all but here's a code review

**Comment by Iamgoofball** [comment]
Not gonna lie, I think the correct move here is to remove all the special code for heels.

**Comment by StrangeWeirdKitten** [comment]
Good solution. Yeah. I was probally going to do this regardless.

**Pull request closed without merging**